File: webapp-net/Core/Models/Entity/LinkList.cs
using Sdl.Web.Common.Models;
using System;
using System.Collections.Generic;

namespace Sdl.Web.Modules.Core.Models
{
    [SemanticEntity(EntityName = "SocialLinks")]
    [Serializable]
    public class LinkList<T> : EntityModel
    {
        public string Headline { get; set; }
        public List<T> Links { get; set; }
        public LinkList()
        {
            Links = new List<T>();
        }
    }
}

File: webapp-java/dxa-module-core/src/main/java/com/sdl/dxa/modules/core/model/entity/LinkList.java
package com.sdl.dxa.modules.core.model.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.sdl.webapp.common.api.model.entity.Link;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public class LinkList extends AbstractEntityModel {

    @JsonProperty("Headline")
    private String headline;

    @JsonProperty("Links")
    private List<Link> links;
}
